.github/workflows/ci.yml

```diff
@@ -54,6 +54,26 @@ jobs:
       - name: check reminders and notify
         uses: agrc/reminder-action@v1
 
+  lockfiles:
+    runs-on: ubuntu-latest
+    name: "Lockfiles"
+    env: 
+      TARGET_DOCS_LOCKFILE_PATH: .lockfiles/py310-dev.lock
+      DOCS_LOCKFILE_PATH: /tmp/lockfile
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        id: setup-python
+        with:
+          python-version: '3.10'
+      - name:  Verify lockfile integrity 
+        run: |
+          pip install tox-uv
+          tox -e docs-lockfile 
+          if ! diff --ignore-matching-lines="^\#" $DOCS_LOCKFILE_PATH $TARGET_DOCS_LOCKFILE_PATH >/dev/null; then
```

Comment:
Hey @Scienfitz, just rebased the PR on main for merge but then noticed that the lockfile pipeline failed. And this made me realize that I made a mistake in my thought process: When generating the lockfile in ci, it will of course always take the latest versions of the secondary dependencies that are compatible with our specified primary dependencies, meaning the the diff will fail whenever there is a secondary dep update. That is, the old lock would still ensure consistent doc creation, but there is also another possible configuration of deps that could have ended up in the lock file.

So I think we need to change the logic. Do you have any suggestion? Probably a reasonable approach is to to apply the same idea as for the changelog: check if pyproject.toml has updated and, if so, show a failed pipeline as a reminder that also the lockfile needs to be updated. Thoughts?

Reply:
well that is what I meant the check will just always fail?
caching the toml seems like a compromise, but perhaps it doesnt belong in the CI at all, we just need a tox env that we can use to save the locks whenever we release

Reply:
So it seems like it is unclear whether the logic implemented here will actually be what we use? Then I'll skip reviewing for now, please ping me once the new logic has been implemented.

Reply:
The won't "always" fail, but they will whenever a dependency that is not pinned to an exact version has a new release that is compatible with our version specs.

But I agree: how about simply following the same approach as with the Changelog then? That is, we can change the pipeline step to implement the following logic: whenever pyproject.toml is changed but the lockfile is NOT changed --> fail. This would serve as the reminder. Not 100% perfect, but sufficient for now, I think

Reply:
Hi @AVHopp, @Scienfitz, quick update on this PR:

I've tried several things but finally implemented the solution mentioned above, i.e. we do the same as for the changelog: if pyproject.toml has changed compared to main, it **can** indicate that the lockfile needs an update and thus the pipeline fails as a reminder to double-check, which can be done with a quick lockfile re-generation locally using the tox command.

The nicer approach – which I tried and failed – would be to regenerate the lock in CI and see if something has changed. This would then only alert us if things really require an update. In principle, this should be possible using the `exclude-newer` option that can be passed to `uv pip compile` and by which you can specify a date to tell uv to only use packages that were available prior to that date. By providing the datetime of the commit where the lockfile was authored (not commited!), one can verify if a change is necessary. The problem was that when executed on CI, there was some additional information in the created file (even though the dependencies remained identical), which produced a non-zero diff. 

So for now, let's simply try with the changelog approach, unless you have a better idea.